source/base/data_out_base.cc

```diff
@@ -7741,19 +7746,61 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
                                   vtk_flags,
                                   ss);
 
-    ierr = MPI_File_write_ordered(
-      fh, ss.str().c_str(), ss.str().size(), MPI_CHAR, MPI_STATUS_IGNORE);
+    // using prefix sum to find specific offset to write at.
+    const std::uint64_t size_on_proc = ss.str().size();
+    std::uint64_t prefix_sum = 0;
+    ierr =
+      MPI_Exscan(&size_on_proc, &prefix_sum, 1, MPI_UINT64_T, MPI_SUM, comm);
     AssertThrowMPI(ierr);
+
+    // Locate specific offset for each processor.
+    const MPI_Offset offset = static_cast<MPI_Offset>(header_size) + prefix_sum;
+
+    ierr =
+      Utilities::MPI::LargeCount::MPI_File_write_at_all_c(fh,
+                                                          offset,
+                                                          ss.str().c_str(),
+                                                          ss.str().size(),
+                                                          MPI_CHAR,
+                                                          MPI_STATUS_IGNORE);
+    AssertThrowMPI(ierr);
+
+    // Sending the offset for writing the footer to rank 0.
+    if (myrank == n_ranks - 1)
+      {
+        footer_offset = size_on_proc + offset;
+        AssertThrowMPI(ierr);
+      }
   }
 
+  if (myrank == 0 || myrank == n_ranks - 1)
+    {
+      ierr = MPI_Sendrecv_replace(&footer_offset,
+                                  1,
+                                  MPI_UINT64_T,
+                                  n_ranks - 1,
+                                  0,
+                                  0,
+                                  0,
+                                  comm,
+                                  MPI_STATUS_IGNORE);
+      AssertThrowMPI(ierr);
+    }
+
   // write footer
   if (myrank == 0)
```

Comment:
Just write the footer on rank ``n_ranks-1`` instead of 0.

Reply:
That fixed version passed the test on my machine so I pushed it! Thank you so much!!